Automattic/jetpack
# Social: Add 'Save to media library' button for SIG image

> Now that we're [changing the logic around attached media](https://github.com/Automattic/jetpack/pull/37964), it's not possible to upload the SIG created image with the social post. As a work around, and a feature that could be useful for other reasons, it would be good to add a 'Save to media library' button, either in the sidebar, or in the SIG options modal, or both. This could then be selected as the attached media.

## Proposed changes:

* Social: Add 'Save to media library' button for SIG image

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* On a site with the paid Social plan, goto post editor
* Turn ON "Enable Social Image"
* Confirm that you see the "Save to media library" button when the image is ready
* Click on the button
* Confirm that it shows, "Saving..." while the request is in progress
* Confirm that the image is successfully saved to the media library

<img width="284" alt="image" src="https://github.com/Automattic/jetpack/assets/18226415/7aaf60a8-f22a-434f-919b-c8422fcd0127">

<img width="291" alt="Screenshot 2024-06-28 at 11 08 47 AM" src="https://github.com/Automattic/jetpack/assets/18226415/08e6c21c-d563-490e-af48-b9e49e07eedd">


https://github.com/Automattic/jetpack/assets/18226415/4cb4e5d7-5121-4f6a-bf33-d49438681cd2
